jupyter_scheduler/executors.py

```diff
@@ -221,3 +222,66 @@ def execute(self):
             archive_filepath = self.staging_paths["tar.gz"]
             with fsspec.open(archive_filepath, "wb") as f:
                 f.write(fh.getvalue())
+
+
+class AllFilesArchivingExecutionManager(DefaultExecutionManager):
+    """Execution manager that archives all output files in and under the
+    output directory into a single archive file
+
+    Notes
+    -----
+    Should be used along with :class:`~jupyter_scheduler.scheduler.AllFilesArchivingScheduler`
+    as the `scheduler_class` during jupyter server start.
+    """
```

Comment:
There is a lot of similar code to `ArchivingExecutionManager`. I found that this made the code much more difficult to read, as I had to switch between the two implementations to discern the differences in behavior. I would highly encourage `AllFilesArchivingExecutionManager` to inherit from `ArchivingExecutionManager` and use both parent private methods and `super().execute()` to inherit logic and minimize code duplication.